AI/VCAI/MapObjectsEvaluator.cpp

```diff
@@ -68,8 +70,47 @@ boost::optional<int> MapObjectsEvaluator::getObjectValue(const CGObjectInstance
 		}
 		return aiValue;
 	}
-	else
-		return getObjectValue(obj->ID, obj->subID);
+	else if(obj->ID == Obj::ARTIFACT)
```

Comment:
A function start to grow. `getObjectValue` should be moved to `CGObjectInstance` as virtual method